CTFd/api/v1/challenges.py

```diff
@@ -102,24 +102,26 @@ def _build_solves_query(extra_filters=(), admin_view=False):
         freeze_cond = true()
     # Finally, we never count solves made by hidden or banned users/teams, even
     # if we are an admin. This is to match the challenge detail API.
-    AccountModel = get_model()
     exclude_solves_cond = and_(
         AccountModel.banned == false(), AccountModel.hidden == false(),
     )
     # This query counts the number of solves per challenge, as well as the sum
     # of correct solves made by the current user per the condition above (which
     # should probably only be 0 or 1!)
     solves_q = (
-        db.session.query(
-            Solves.challenge_id,
-            sa_func.count(Solves.challenge_id),
-            sa_func.sum(cast(user_solved_cond, sa_types.Integer)),
-        )
+        db.session.query(Solves.challenge_id, sa_func.count(Solves.challenge_id),)
         .join(AccountModel)
         .filter(*extra_filters, freeze_cond, exclude_solves_cond)
         .group_by(Solves.challenge_id)
     )
-    return solves_q
+    # Also gather the user's solve items which can be different from above query
+    # Even if we are a hidden user, we should see that we have solved the challenge
+    # But as a hidden user we are not included in the count
+    solve_ids = (
+        Solves.query.with_entities(Solves.challenge_id).filter(user_solved_cond).all()
+    )
```

Comment:
This probably needs to respect `freeze_cond, *extra_filters` as well

Reply:
Perhaps we should create a list like `solve_filters = [*extra_filters, freeze_cond, exclude_solves_cond]` to avoid drift between the two queries

Reply:
Off of the top of my head I'm not sure... I believe this is for calculating the user's solves which should always be shown regardless of freeze time. For example, I can solve a challenge during freeze time and I should be able to see that, but other teams shouldn't be able to see that I have solved it. 

But not totally sure, I will confirm the behavior later today.

Reply:
Right, I vaguely thought that might be the case but wasn't sure what the expected behaviour should be. I guess the visible effects of this would be:
 * solve counts will only include solves which can contribute to final scores
 * solves outside of the freeze time will properly mark challenges on the board green so the player (and JS) can know not to re-attempt it
 * since the `solves/` endpoint should respect freeze time, solves outside the freeze time will match the score count in the challenge list and details API endpoints

Sounds alright to me :+1:

Reply:
I think there is an issue where a user's profile page (`/user`, `/profile`) doesn't show solves during freeze time. Might be a source of confusion here. 

I think that's incorrect and we should always show the user's solves to them. Don't need to fix it in this PR but I'll probably fix it in the next one.

Reply:
I believe that I can confirm that we do not need to consider freeze time in the second query. 

It should always return the user's solves. 

I will check for other edge cases like anonymous users, add a clearer docstring and then merge.

Reply:
Actually I'm not even sure if we should show a user's solves in their profile during freeze time.... I'll think about it before releasing 3.3.